### PR TITLE
feat: save shipping address and contact to woocommerece customer

### PR DESCRIPTION
**Pull request by scmmishra**
This PR makes the following changes
1. Solves: https://github.com/frappe/erpnext/issues/21481
1. Uses customer name to create and link addresses, this is to fix duplicate entry error if a two customers with same name but different email address place an order

To test 1, you can run the `test_sales_order_for_woocommerce` method. For testing the renaming change the email address in the data and run the test again

### CODE REVIEW

**Comment by scmmishra** [comment]
@Mergifyio backport version-12-hotfix

File: erpnext/erpnext_integrations/connectors/woocommerce_connection.py [hunk 1]
```diff
@@ -68,38 +69,48 @@ def link_customer_and_address(raw_billing_data, customer_name):
...
+	address.pincode = raw_data.get("postcode")
+	address.phone = raw_data.get("phone")
+	address.email_id = customer.woocommerce_email
 	address.append("links", {
 		"link_doctype": "Customer",
 		"link_name": customer.customer_name
 	})
+
 	address.flags.ignore_mandatory = True
 	address = address.save()
```
Comment (Mangesh-Khairnar) [inline erpnext/erpnext_integrations/connectors/woocommerce_connection.py]: Not returning anything, check line 79, 80

**Review by Mangesh-Khairnar — CHANGES_REQUESTED** [review]

File: erpnext/erpnext_integrations/connectors/woocommerce_connection.py [hunk 2]
```diff
@@ -68,38 +69,80 @@ def link_customer_and_address(raw_billing_data, customer_name):
...
+
+	contact.append("links", {
+		"link_doctype": "Customer",
+		"link_name": customer.name
+	})
+
+	contact.flags.ignore_mandatory = True
+	contact.save()
+
+	return contact
```
Comment (Mangesh-Khairnar) [inline erpnext/erpnext_integrations/connectors/woocommerce_connection.py]: no point in returning the contact?

File: erpnext/erpnext_integrations/connectors/woocommerce_connection.py [hunk 3]
```diff
@@ -68,38 +69,80 @@ def link_customer_and_address(raw_billing_data, customer_name):
...
-	address.woocommerce_email = customer_woo_com_email
-	address.address_type = "Billing"
-	address.country = frappe.get_value("Country", {"code": raw_billing_data.get("country", "IN").lower()})
-	address.state = raw_billing_data.get("state")
-	address.pincode = raw_billing_data.get("postcode")
-	address.phone = raw_billing_data.get("phone")
-	address.email_id = customer_woo_com_email
+		billing_address = create_address(raw_billing_data, customer, "Billing")
+		shipping_address = create_address(raw_shipping_data, customer, "Shipping")
+		contact = create_contact(raw_billing_data, customer)
```
Comment (Mangesh-Khairnar) [inline erpnext/erpnext_integrations/connectors/woocommerce_connection.py]: contact variable is not used anywhere

**Review by Mangesh-Khairnar — CHANGES_REQUESTED** [review]

**Review by Mangesh-Khairnar — APPROVED** [review]

**Pull request merged**